lib/capsule_crm/client.rb

```diff
@@ -0,0 +1,91 @@
+module CapsuleCrm
+  # other unexpected errors
+  class ApiFailure < StandardError; end
+  # 400 incorrect request body
+  class BadRequest < ApiFailure; end
+  # 401 invalid api key
+  class NotAuthorised < ApiFailure; end
+  # 403 operation not permitted
+  class NotAllowed < ApiFailure; end
+  # 404 incorrect url
+  class NotFound < ApiFailure; end
+  # 422 field validation failed
+  class ValidationFailed < ApiFailure; end
+
+  class Client
+    BASE_URL = 'https://api.capsulecrm.com/api/v2'.freeze
+
+    def initialize(api_key: ENV['CAPSULECRM_API_KEY'], connection: nil)
+      @api_key = api_key
+      @connection = connection || Faraday.new(BASE_URL, headers: headers)
+    end
+
+    def milestones
+      get_data('/milestones')
+    end
+
+    def field_definitions(entity = :parties)
+      get_data("/#{entity}/fields/definitions")
+    end
+
+    def users
+      get_data('/users')
+    end
+
+    def create_party(party)
+      post_data('/parties', party)
+    end
+
+    def create_opportunity(opportunity)
+      post_data('/opportunities', opportunity)
+    end
+
+    private
+
+    def get_data(path, params = nil)
+      response = @connection.get("#{BASE_URL}#{path}", params)
+      return JSON.parse(response.body) if response.success?
```

Comment:
wonder about the built json per https://lostisland.github.io/faraday/#/middleware/included/json